### ISSUE TITLE
os.exit() call does not trigger shutdown hook

### ISSUE THREAD

**Issue by heisenbugh** [brndnmtthws/conky#386]
Hi,
Acording to my logic I have to exit conky process depending on some condition. For this purpose I use "os.exit()" function call in lua. This call closes conky window(process) but does not trigger "lua_shutdown_hook". I have to do some shutdown operations.
Am I doing something wrong or is there any proper way doing this(exiting conky also triggering shutdown hook)?

Thank you.

**Comment by plikhari** [comment]
You do not need to give any 'os.exit' in lua script run within conky - when a conky process is shut down - all internal sub processes also shut down.

conky interacts directly with lua using tolua++ system (this happens during compile time)- in other words - lua is integrated with conky and no separate lua interpreter is launched.

That is the reason you do not need to think of it as if you are running a lua process like if you would be running a lua script from the command prompt.

What shut down operations do you need to do - perhaps I can help.

It is a good idea to run conky from a script and in that script you can use the following line before you run any conky trigger lines
```
if [[ $(pidof conky) ]] ; then killall -SIGUSR1 conky; fi
```

**Comment by heisenbugh** [comment]
Currently i changed my logic to control exit operations by an external script (not sure but i guess as you suggested.)

I try to do a drive monitoring tool. If a drive is mounted then a conky instance will run and display drive info(free space, capacity...) And then if drive is unmounted conky instance have to be killed(exited.) I had placed this condition logic in a lua file which is loaded by conky. I control instance creation by a script, but for exiting I used os.exit() within conky instance ( conky instance decided whether it has to exit or continue). I need some clean up operations before exiting (removing some tmp files...) and put this cleaning in to shutdown hook. But shutdown hook does not triggered after os.exit. 
Also I can write a wrapper function OsExit()  (may be if I need I can do something like this):
`function shutdownhook()`
`    Cleanup()`
`end`
`function OsExit()`
`    shutdownhook()`
`    os.exit()`
`end`

Thank you very much.
